### ISSUE TITLE
ThreadElfCore.h: fails to build with `error: static assertion failed due to requirement`

### ISSUE THREAD

**Issue by sylvestre** [llvm/llvm-project#117327]
on linux i386: 
```
FAILED: tools/lldb/source/Plugins/Process/elf-core/CMakeFiles/lldbPluginProcessElfCore.dir/ThreadElfCore.cpp.o 
/opt/sccache//sccache /build/source/build-llvm/./bin/clang++ -DHAVE_ROUND -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_LARGEFILE_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/source/Plugins/Process/elf-core -I/build/source/lldb/source/Plugins/Process/elf-core -I/build/source/lldb/include -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/include -I/build/source/build-llvm/tools/clang/stage2-bins/include -I/build/source/llvm/include -I/usr/include/python3.11 -I/build/source/clang/include -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/../clang/include -I/build/source/lldb/source -I/build/source/build-llvm/tools/clang/stage2-bins/tools/lldb/source -isystem /usr/include/libxml2 -fstack-protector-strong -Wformat -Werror=format-security -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -ffile-prefix-map=/build/source/build-llvm/tools/clang/stage2-bins=../../../../ -ffile-prefix-map=/build/source/= -no-canonical-prefixes -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-vla-extension -O2 -DNDEBUG -g1  -fno-exceptions -funwind-tables -std=c++17 -MD -MT tools/lldb/source/Plugins/Process/elf-core/CMakeFiles/lldbPluginProcessElfCore.dir/ThreadElfCore.cpp.o -MF tools/lldb/source/Plugins/Process/elf-core/CMakeFiles/lldbPluginProcessElfCore.dir/ThreadElfCore.cpp.o.d -o tools/lldb/source/Plugins/Process/elf-core/CMakeFiles/lldbPluginProcessElfCore.dir/ThreadElfCore.cpp.o -c /build/source/lldb/source/Plugins/Process/elf-core/ThreadElfCore.cpp
In file included from /build/source/lldb/source/Plugins/Process/elf-core/ThreadElfCore.cpp:44:
/build/source/lldb/source/Plugins/Process/elf-core/ThreadElfCore.h:120:15: error: static assertion failed due to requirement 'sizeof(ELFLinuxSigInfo) == 56': sizeof ELFLinuxSigInfo is not correct!
  120 | static_assert(sizeof(ELFLinuxSigInfo) == 56,
      |               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/lldb/source/Plugins/Process/elf-core/ThreadElfCore.h:120:39: note: expression evaluates to '44 == 56'
  120 | static_assert(sizeof(ELFLinuxSigInfo) == 56,
      |               ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
1 error generated.
```

**Comment by sylvestre** [comment]
Regression range: 68ce528defae..dde9477d8c0b

**Comment by Jlalond** [comment]
@sylvestre closing because this was fixed in [4ab298b](https://github.com/llvm/llvm-project/commit/4ab298b5fbc8f48387062b2dd99ea07127c02e6b)
